### PR TITLE
Cleaning up some regex expressions

### PR DESCRIPTION
**Pull request by mbeltagy**
The or-ing `r"^\s*(?:#\s*(.*?)\s*$|$)"` seemed a bit redundant 
a `?` coming after `.*` is also redundant

### CODE REVIEW

**Comment by waldyrious** [comment]
> a `?` coming after `.*` is also redundant

That's actually a thing in regex. It's called a ["lazy"](https://www.regular-expressions.info/repeat.html) modifier.

**Comment by mbeltagy** [comment]
Indeed.  But in the context of the example,  it does not do anything.

File: doc/src/manual/strings.md [hunk 1]
```diff
@@ -830,10 +830,10 @@ x   Tells the regular expression parser to ignore most whitespace
 For example, the following regex has all three flags turned on:
 
 ```jldoctest
-julia> r"a+.*b+.*?d$"ism
+julia> r"a+.*b+.*d$"ism
 r"a+.*b+.*?d$"ims
```
Comment (Keno) [inline doc/src/manual/strings.md]: Need to change the output here as well

File: doc/src/manual/strings.md [hunk 2]
```diff
@@ -681,7 +681,7 @@ of the substring that matches, but perhaps we want to capture any non-blank text
 character. We could do the following:
 
 ```jldoctest
-julia> m = match(r"^\s*(?:#\s*(.*?)\s*$|$)", "# a comment ")
+julia> m = match(r"^\s*(?:#\s*(.*?)\s*$)", "# a comment ")
```
Comment (vtjnash) [inline doc/src/manual/strings.md]: this changes the meaning of regex ("\s*" matches the former and not the latter), although I don't know if the difference is material (the regex is only a rough approximation of the operation it describes, and would be tripped up by strings, character literals, or multi-line comments)

**Comment by vtjnash** [comment]
> a ? coming after .* is also redundant

It's not redundant primarily because of that, but due to the presence of `$` in this example (and the lack of any capturing groups to create or observe any differences)

**Review by vtjnash — APPROVED** [review]

File: doc/src/manual/strings.md (hunk 2 above)
Comment (mbeltagy) [reply doc/src/manual/strings.md]: The `*` after the `\s` is for zero or more, so the `$|$` alternation will never be of use.

**Comment by fingolfin** [comment]
This was approved in May 2018, and still applies. I've triggered a fresh CI run and hopefully we can merge it afterwards... We'll see

**Pull request merged**